module/behaviour/strategy/PS3Walk/src/PS3Walk.cpp

```diff
@@ -17,60 +17,203 @@
+                                break;
+                            case JoystickEvent::Button::DPAD_LEFT:
+                                event->button = JoystickEvent::Button::DPAD_LEFT;
+                                break;
+                            case JoystickEvent::Button::DPAD_RIGHT:
+                                event->button = JoystickEvent::Button::DPAD_RIGHT;
+                                break;
+                        }
+                    }
+                    if (is_axis) {
+                        switch (event_buffer.front()) {
+                            case JoystickEvent::Axis::LEFT_JOYSTICK_HORIZONTAL:
+                                event->axis = JoystickEvent::Axis::LEFT_JOYSTICK_HORIZONTAL;
+                                break;
+                            case JoystickEvent::Axis::LEFT_JOYSTICK_VERTICAL:
+                                event->axis = JoystickEvent::Axis::LEFT_JOYSTICK_VERTICAL;
+                                break;
+                            case JoystickEvent::Axis::L2: event->axis = JoystickEvent::Axis::L2; break;
+                            case JoystickEvent::Axis::RIGHT_JOYSTICK_HORIZONTAL:
+                                event->axis = JoystickEvent::Axis::RIGHT_JOYSTICK_HORIZONTAL;
                                 break;
-                            case AXIS_RIGHT_JOYSTICK_VERTICAL: headPitch = -event.value; break;
-                            case AXIS_RIGHT_JOYSTICK_HORIZONTAL: headYaw = -event.value; break;
+                            case JoystickEvent::Axis::RIGHT_JOYSTICK_VERTICAL:
+                                event->axis = JoystickEvent::Axis::RIGHT_JOYSTICK_VERTICAL;
+                                break;
+                            case JoystickEvent::Axis::R2: event->axis = JoystickEvent::Axis::R2; break;
                         }
                     }
-                    else if (event.isButton()) {
-                        // event was a button event
-                        switch (event.number) {
-                            case BUTTON_TRIANGLE:
+
+                    if (is_init) {
+                        // TODO: Do we need to handle this?
+                    }
+
+                    if (is_other) {
+                        log<NUClear::WARN>("Unknown event on joystick. Ignoring");
+                    }
+
+                    // Erase axis/button from buffer
+                    event_buffer.erase(event_buffer.begin(), std::next(event_buffer.begin(), sizeof(uint8_t)));
+
+                    if (!is_other) {
+                        emit(std::move(event));
+                    }
+                }
+            });
+
+            // Trigger when the joystick accelerometer has an event to read
+            on<IO>(joystick_acc_fd, IO::READ).then([this] {
+                // Accelerometer comes in here .... if you know which axis is which
+            });
+
+            on<Trigger<JoystickEvent>>().then([this](const JoystickEvent& event) {
```

Comment:
You should move this to another module so that people can still use the controller to test various other things without having to have an entire walk/wall controller coming along with it.